ml/CV/Hemo_RNN_CV.py

```diff
@@ -0,0 +1,186 @@
+                buffer_size=10000,
+                rnn_units=64,
+                hidden_dim=64,
+                embedding_dim=32,
+                reg_strength=0,
+                lr=1e-3,
+                drop_rate=0.1
+               )
+
+
+def counts_aa(vec):
+    counts =  tf.histogram_fixed_width(vec, [0, 20], nbins=21)[1:]
+    return counts/tf.reduce_sum(counts)
+
+
+describtion  = 'baseline_counts_fr_removed_first_and_second_dens_group'
+
+def build_model(L):
+    inputs = tf.keras.Input(shape=(L,))
+    input_f = tf.keras.Input(shape=(20,))
+    # make embedding and indicate that 0 should be treated as padding mask
+    e = tf.keras.layers.Embedding(input_dim=config.vocab_size, 
+                                        output_dim=config.embedding_dim,
+                                        mask_zero=True)(inputs)
+
+    # RNN layer
+    x = tf.keras.layers.Bidirectional(tf.keras.layers.LSTM(config.rnn_units, return_sequences=True))(e)
+    x = tf.keras.layers.Bidirectional(tf.keras.layers.LSTM(config.rnn_units))(x)
+    x = tf.keras.layers.Concatenate()([x, input_f])
+    x = tf.keras.layers.LayerNormalization()(x)
+    x = tf.keras.layers.Dropout(config.drop_rate)(x)
+    # a dense hidden layer
+#     x = tf.keras.layers.Dense(
+#         config.hidden_dim, 
+#         activation='relu', 
+#         kernel_regularizer=tf.keras.regularizers.l2(config.reg_strength))(x)
+#     x = tf.keras.layers.Concatenate()([x, input_f])
+#     x = tf.keras.layers.LayerNormalization()(x)
+#     x = tf.keras.layers.Dropout(config.drop_rate)(x)
+#     x = tf.keras.layers.Dense(
+#         config.hidden_dim // 4, 
+#         activation='relu', 
+#         kernel_regularizer=tf.keras.regularizers.l2(config.reg_strength))(x)
+#     x = tf.keras.layers.Concatenate()([x, input_f])
+#     x = tf.keras.layers.LayerNormalization()(x)
+#     x = tf.keras.layers.Dropout(config.drop_rate)(x)
+    # predicting prob, so no activation
+    yhat = tf.keras.layers.Dense(1, activation='sigmoid')(x) # 
+
+    model = tf.keras.Model(inputs=[inputs, input_f], outputs=yhat, name='hemo-rnn')
+    return model
+
+if use_tpu:
+  tf.keras.backend.clear_session()
+
+  resolver = tf.distribute.cluster_resolver.TPUClusterResolver('grpc://' + os.environ['COLAB_TPU_ADDR'])
+  tf.config.experimental_connect_to_cluster(resolver)
+
+  # This is the TPU initialization code that has to be at the beginning.
+  tf.tpu.experimental.initialize_tpu_system(resolver)
+  print("All devices: ", tf.config.list_logical_devices('TPU'))
+
+  strategy = tf.distribute.experimental.TPUStrategy(resolver)
+
+i = np.arange(len(labels))
+np.random.shuffle(i)
+shuffled_labels = labels[i]
+shuffled_features = features[i]
+ 
+L = None#features[0].shape[-1]
+N = len(shuffled_labels)
+
+from sklearn.model_selection import ShuffleSplit
+kf = KFold(n_splits=5, shuffle=True)
+folds_results = []
+for i, (train_index, test_index) in enumerate(kf.split(shuffled_features, shuffled_labels)):
```

Comment:
@whitead This is the CV part. The rest of the script is quite similar to what is already in the notebooks.